src/State/RegionState.lua

```diff
@@ -95,6 +95,29 @@ function RegionState:AddRegion(RegionName: string, Center: CFrame, Size: Vector3
     })
 end
 
+--[[
+Creates regions based on a pre-determined model/folder, provides for quicker setup &
+easier customization. 
+Name each region with the appropriate title and then add parts to determine their region.
+--]]
+function RegionState:InsertRegionsFromInstance(Instances: Model | Folder | Instance): ()
+	if #Instances:GetChildren() == 0 then
+		error("Instance \"" .. Instances:GetFullName() .. "\" has no child instances, unable to build Regions from it.")
+	end
+	for _, RegionParent in Instances:GetChildren() do 
+		--Get the region name from the child name
+		local RegionName = RegionParent.Name
+		for _, RegionPart in RegionParent:GetChildren() do 
+			if RegionPart:IsA("BasePart") then
+				self:AddRegion(RegionName, RegionPart.CFrame, RegionPart.Size)
+			end
+		end
+	end
+	--// TODO
+	--// Perform region checks to maybe auto-connect intersecting regions? 
```

Comment:
I don't really see the need to handle checking for intersections. You may actually want that like.